✨ feat: 스터디 공고 지원 - 지원 내역 목록 조회

Jiwoong0322 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #134 
- 작업 요약: 스터디 공고 지원 - 지원 내역 목록 조회 기능 개발

## 📄 상세 내용
- [ ] 공고 지원자 목록 조회 API 구현 (GET /api/v1/recruitments/{recruitment_uuid}/applications/all)
- [ ] 커서 기반 페이지네이션(DefaultCursorPagination) 기능 적용
- [ ] Application 모델의 __str__ 메소드 동작을 검증하는 단위 테스트 추가
- [ ] 공고 작성자만 목록을 조회할 수 있도록 IsRecruitmentAuthor 권한 추가

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Jiwoong0322 committed:
✨ feat: 스터디공고 지원 - 지원 내역 목록 조회

  1.  공고 지원자 목록 조회 API 구현 (GET /api/v1/recruitments/{recruitment_uuid}/applications/all)
  2.  공고 작성자만 목록을 조회할 수 있도록 IsRecruitmentAuthor 권한 추가
  3.  커서 기반 페이지네이션(DefaultCursorPagination) 기능 적용
  4.  Application 모델의 __str__ 메소드 동작을 검증하는 단위 테스트 추가

관련 이슈: #134

Umdoong commented on apps/applications/urls/recruitments_uuid_urls.py:
```diff
@@ -1,10 +1,16 @@
+    # POST /api/v1/recruitments/{uuid}/applications
     path("", ApplicationAPIView.as_view(), name="application-create"),
+    # GET /api/v1/recruitments/{uuid}/applications/all
+    path("/all", RecruitmentApplicationListView.as_view(), name="recruitment-application-list"),
```
all은 빼주세요! RESTful하지 않습니다.
post와 같은 경로에서 get으로 만들어주시면 될 것 같아요

Umdoong commented on apps/applications/views/applications_list_views.py:
이것도 마찬가지로 하나의 view에서 관리해주세요
그리고 application_views.py와 applications_views.py 파일이 있는데 다른 팀처럼 applications_views 이렇게 맞춰주세용

Umdoong commented on apps/applications/managers/applications_list_managers.py:
이것도 마찬가지입니다.
Application 객체를 다루는 매니저가 이미 있는데 굳이 분리할 필요는 없을 것 같습니다.

Umdoong commented on apps/recruitments/recruitments_permissions.py:
해당 permission을 사용하는 view 혹은 view 내부에서 호출하는 service에서 Recruitment 객체를 사용할 일이 생기면 DB 조회가 중복으로 일어날 수 있을 것 같습니다.
재활용 할 수 있도록 하는 방법을 고민해보시면 좋을 것 같아요

Jiwoong0322 committed:
♻️ refactor: applications 파일 통합

1. API 엔드포인트 및 View 구조 개선
 - 지원서 생성와 지원자 목록 조회 API의 엔드포인트 통합
 - 분리되어 있던 View들을 ApplicationAPIView 하나로 통합.
2. 권한 로직 최적화
 - IsRecruitmentAuthor 권한 클래스에서 공고 객체 조회 시, 해당 객체를 request에 저장하도록 하여 View에서 발생하는 중복 DB 조회를 방지하고 성능을 개선
3. 코드 구조 리팩토링 및 정리
 - 여러 개로 분리되어 있던 managers를 하나로 통합, 리팩토링 과정에서 불필요해진 view, manager, service 파일을 삭제
4. 테스트 및 타입 에러 수정
 - 리팩토링으로 인해 변경된 URL 이름, 권한 로직, 누락됨 import등 테스트 코드와 소스 코드에 반영하여 mypy 타입 에러와 테스트 실패 해결.

관련 이슈: #134

# Conflicts:
#	apps/recruitments/urls/recruitments_urls.py

# Conflicts:
#	apps/recruitments/urls/recruitments_urls.py

Jiwoong0322 committed:
♻️ refactor: mypy 에러 해결

1. applications_views.pydml get_permissions 함수에 반환 타입 어노테이션이 누락되있던 문제 해결
2. 들여쓰기 오류 해결

관련 이슈: #134
